### PR TITLE
#2831 use maskedMoneyInput in redeem flow

### PR DESCRIPTION
**Pull request by AlexeyTuboltsev**
refactor maskedMoneyInput to play nicely with formik
refactor BankTransferRedeemInit to use maskedMoneyInput
refactor MoneySuiteWidget to get rid of deprecated number/money formatters

### CODE REVIEW

**Comment by AlexeyTuboltsev** [comment]
canceled for now.

**Comment by AlexeyTuboltsev** [comment]
FOR QA:
this pr rebuilds BankTransferRedeemInit. 
Please check if it works in the same way as it worked before

**Comment by AlexeyTuboltsev** [comment]
1300

File: app/components/shared/formatters/Money.tsx [hunk 1]
```diff
@@ -70,28 +71,28 @@ export const selectCurrencyCode = (moneyFormat: TMoneyFormat): string => {
...
+  className,
 }) => {
   let formattedValue = null;
-
   if (value) {
     //todo: this should pass through 0 as well. Use isValidNumber from the #2687 PR when it's merged
     const decimalPlaces = selectDecimalPlaces(moneyFormat, outputFormat);
     formattedValue =
       outputFormat === EHumanReadableFormat.FULL ||
+      outputFormat === EHumanReadableFormat.ONLY_NONZERO_DECIMALS ||
```
Comment (Moejoe90) [inline app/components/shared/formatters/Money.tsx]: For readability please move this to a constant with a name that explains the reason behind this logic

File: app/modules/bank-transfer-flow/sagas.ts [hunk 2]
```diff
@@ -93,7 +93,7 @@ export function* getRedeemData({ contractsService }: TGlobalDependencies): any {
     minEuroUlps: contractsService.euroTokenController.minWithdrawAmountEurUlps,
   });
 
-  yield put(actions.bankTransferFlow.setRedeemData(bankFeeUlps, minEuroUlps));
+  yield put(actions.bankTransferFlow.setRedeemData(bankFeeUlps, minEuroUlps.toString()));
```
Comment (Moejoe90) [inline app/modules/bank-transfer-flow/sagas.ts]: `setRedeemData` was the type always set to `string`?

**Comment by glupiawymowka** [comment]
I left a comment that somehow vanished: 
It happened to me once, that nEUR to EUR money module showed "0" and reloaded only when I clicked on the page. 
However, I was not able to reproduce it with any other login/ fixture so this is just a side note to keep in mind

**Pull request merged**